xwiki-platform-core/xwiki-platform-repository/xwiki-platform-repository-server-api/src/main/java/org/xwiki/repository/internal/RepositoryManager.java

```diff
@@ -707,6 +735,15 @@ public DocumentReference importExtension(String extensionId, ExtensionRepository
         return document.getDocumentReference();
     }
 
+    public boolean isVersionProxyingEnabled(XWikiDocument extensionDocument)
+    {
+        BaseObject extensionProxyObject = extensionDocument.getXObject(XWikiRepositoryModel.EXTENSIONPROXY_CLASSREFERENCE);
+        if(extensionProxyObject == null){
+            return false;
+        }
+        return "version".equals(getValue(extensionProxyObject, XWikiRepositoryModel.PROP_PROXY_PROXYLEVEL, (String) null));
```

Comment:
Would be nice to put "version" in a field of XWikiRepositoryModel along with the other possible values.

Also something like "history" might be a better name since the last version is not proxied, WDYT ?